src/config-validator.ts

```diff
@@ -0,0 +1,121 @@
+import {z, ZodArray, ZodRecord, ZodString} from "zod";
+import {SchemaInfo} from "./middlewares/datasource";
+
+
+function getRequiredEnvVar(name: string){
+    const property = process.env[name] as any;
+    if(!property || property.trim().length==0){
+        throw new Error("config file is missing property "+name);
+    }
+    return property;
+}
+
+
+export const getRequiredNumber = (name: string): number => {
+    const value = parseInt(getRequiredEnvVar(name));
+    const numberValue = z.number().finite().safeParse(value);
+    if(!numberValue.success) {
+        throw new Error("property "+name+" in config is not a valid number")
+    }
+    return numberValue.data;
+}
+
+
+export const getRequiredString = (name: string): string => {
+    const value = getRequiredEnvVar(name);
+    const stringValue = z.string().safeParse(value)
+    if(!stringValue.success) {
+        throw new Error("property "+name+" in config is not a valid string");
+    }
+    return stringValue.data;
+}
+
+
+export const getString = (name:string): string => {
+    const value = process.env[name] || '';
+    const stringValue = z.string().safeParse(value)
+    if(!stringValue.success) {
+        throw new Error("property "+name+" in config is not a valid string");
+    }
+    return stringValue.data;
+}
+
+
+export const getUrl = (name:string): string => {
+    const value = process.env[name] || 'a://';
```

Comment:
the default value for the url was intended to avoid the need to add the egoUrl property in case the service is configured to have no authentication at all.

Reply:
Perhaps we need some mechanism to indicate if Ego auth is used or not. If it is used, this value is mandatory. If not, we never check this value so its not required.

This can be done using an environment variable to define an auth profile or mechanism. It can be set to `NO_AUTH` or to `EGO` (or in the future, perhaps also `OAUTH` or `KEYCLOAK`). With this we can then apply the correct checks based on the profile being selected.

Reply:
the env property `AUTH_STRATEGY` does exactly that. It can be set to `EGO`, `KEYCLOAK` or `NONE`